### PR TITLE
Jetpack Social: Add the remaining social shares view to the post settings screen

### PR DESCRIPTION
**Pull request by wargcm**
See: #20784

## Description

Adds the remaining shares view to the post settings screen.

## Screenshots

| Remaining Shares - Light | Remaining Shares - Dark |
|------|------|
| Light remaining |  Dark remaining |

| No Remaining Shares - Light | No Remaining Shares - Dark |
|------|------|
| Light no remaining | Dark no remaining |

## Testing

To test:
- Launch Jetpack and login
- Select a site with a social connection
- Create a blog post or edit one
- On the editor, open the menu (`...`) in the top right
- Tap on `Post Settings`
- **Verify** the remaining shares UI matches: Th1ahHKq53k5JT1PNDMavY-fi-1043%3A15560
- Update `displayWarning` in the view model to default to `true`
- Relaunch and navigate back to the post settings screen
- **Verify** the UI with the warning matches: Th1ahHKq53k5JT1PNDMavY-fi-1043%3A16716

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import SwiftUI
+
+struct JetpackSocialRemainingSharesView: View {
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift]: A bit of a nitpick, but it looks like this view is specific to Post Settings, and there's going to be another one in the pre-publishing sheet. What do you think about adding "Post Settings" in the class name? e.g.

```
JetpackSocialPostSettingsRemainingSharesView
```

It is kinda long, though...

File: WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        let sharesRemaining = Text(sharesRemainingString).font(.callout)
+        if viewModel.displayWarning {
+            return sharesRemaining
+        }
+        let remainingDays = Text(Constants.remainingEndText).font(.callout).foregroundColor(.secondary)
+        return sharesRemaining + remainingDays
+    }
+
+    private struct Constants {
+        static let remainingTextFormat = NSLocalizedString("social.remainingshares.text.format",
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift]: Same comment above applies here too: should we add "post settings" here to hint that this string is only displayed in Post Settings screen?

File: WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+struct JetpackSocialRemainingSharesViewModel {
+    let remaining: Int
+    let limit: Int
+    let displayWarning: Bool
+    let onSubscribeTap: () -> Void
+
+    init(remaining: Int = 27,
+         limit: Int = 30,
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift]: are these defaults for testing purposes? I'm thinking we might not need to provide any defaults here 🤔.

**Review by dvdchr — APPROVED** [review]
Left some comments, but changes are looking good to me.

File: WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift (hunk 1 above)
Comment (wargcm) [reply WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift]: We can probably shorten it a tiny bit by removing `Post` so: `JetpackSocialSettingsRemainingSharesView`. Do you think that it's still clear enough?

File: WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift (hunk 3 above)
Comment (wargcm) [reply WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift]: Yep, these are just here so it's quick and easy to mess with the UI settings. They will likely go away when we actually pass the real values to this UI.

File: WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift (hunk 1 above)
Comment (dvdchr) [reply WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift]: Looks good to me!

Comment (wargcm) [reply WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialRemainingSharesView.swift]: Awesome! I renamed it with this commit along with the string keys: https://github.com/wordpress-mobile/WordPress-iOS/pull/20966/commits/6ed7c0933781a3fbcd4ac53f37e1dd12d1cdf7d7

**Pull request merged**